src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs

```diff
@@ -1293,6 +1295,318 @@ public async Task When_Large_List_Scroll_To_End_Then_Back_Up_And_First_Item()
+					var border = new Border()
+					{
+						Height = 50,
+						Child = tb
+					};
+
+					return border;
+				})
+			};
+			container.Children.Add(list);
+
+			var source = new List<string>();
+
+			var random = new Random(42);
+			string RandomString(int length)
+			{
+				const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+				return new string(Enumerable.Repeat(chars, length)
+					.Select(s => s[random.Next(s.Length)]).ToArray());
+			}
+
+			for (var i = 0; i < 100; i++)
+			{
+				source.Add(RandomString(5));
+			}
+			list.ItemsSource = source;
+
+			WindowHelper.WindowContent = container;
+			await WindowHelper.WaitForIdle();
+
+			ScrollTo(list, 1000000); // Scroll to end
+
+			await Task.Delay(200);
+			await WindowHelper.WaitForIdle();
+
+			ScrollTo(list, 50); // scroll back up but not all the way
+			ScrollTo(list, 0);
+
+			await Task.Delay(200);
+			await WindowHelper.WaitForIdle();
+
+			var firstContainer = (FrameworkElement)list.ContainerFromIndex(0);
+
+			firstContainer.Should().NotBeNull();
+			LayoutInformation.GetLayoutSlot(firstContainer).Y.Should().BeLessOrEqualTo(0);
+
+			var secondContainer = (FrameworkElement)list.ContainerFromIndex(1);
+
+			secondContainer.Should().NotBeNull();
+			LayoutInformation.GetLayoutSlot(secondContainer).Y.Should().Be(50);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if !__SKIA__
+		[Ignore("InputInjector is only supported on skia")]
+#endif
+		public async Task When_Large_List_Scroll_To_End_Then_Back_Up_TryClick()
+		{
+			var container = new Grid { Height = 500, Width = 100 };
+
+			var list = new ListView
+			{
+				ItemTemplate = new DataTemplate(() =>
+				{
+
+					var tb = new TextBlock();
+					tb.SetBinding(TextBlock.TextProperty, new Binding());
+					var border = new Border()
+					{
+						Height = 50,
+						Child = tb
+					};
+
+					return border;
+				})
+			};
+			container.Children.Add(list);
+
+			var source = new List<string>();
+
+			var random = new Random(42);
+			string RandomString(int length)
+			{
+				const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+				return new string(Enumerable.Repeat(chars, length)
+					.Select(s => s[random.Next(s.Length)]).ToArray());
+			}
+
+			for (var i = 0; i < 100; i++)
+			{
+				source.Add(RandomString(5));
+			}
+			list.ItemsSource = source;
+
+			WindowHelper.WindowContent = container;
+			await WindowHelper.WaitForIdle();
+
+			var injector = InputInjector.TryCreate() ?? throw new InvalidOperationException("Failed to init the InputInjector");
+			var mouse = injector.GetMouse();
+
+			var bounds = list.GetAbsoluteBounds();
+			mouse.MoveTo(new Point(bounds.GetMidX(), bounds.Top + 10));
+			mouse.Press();
+			mouse.Release();
+
+			await Task.Delay(200);
+			await WindowHelper.WaitForIdle();
+
+			// This problem is incredibly difficult to reproduce. These are the exact
+			// mouse wheel deltas that my trackpad produced when I reproduced this manually.
+			var deltasString = """
```

Comment:
You can use a `new int[]{ -266, -393, ... }` instead, this will avoid typos and will be foldable in VS.